### PR TITLE
BlazeMeter Cloud Trial Image

### PR DESCRIPTION
**Pull request by ztreinhart**
Trial image to provide integration with Blazemeter.

### CODE REVIEW

**Comment by zeisss** [comment]
Structure looks good - any idea how to handle the metrics?

**Comment by ztreinhart** [comment]
This needs a little tweak still to make sure Taurus writes the summary metrics CSV file to the proper location. Other than that it's very close.

**Comment by ztreinhart** [comment]
This works now, at least in my testing. Just need to finish up the README and add at least a rudimentary test.

File: blazemeter-cloud/docker-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	exit 1
+fi
+
+if [ -z "${BLAZEMETER_API_SECRET}" ]; then
+	echo "ERROR: BLAZEMETER_API_SECRET not found. An API ID and an API Secret are required to access Blazemeter." > /dev/stderr
+	exit 1
+fi
+
+if [ -z "${BLAZEMETER_TEST_URL}" ]; then
+	echo "ERROR: BLAZEMETER_TEST_URL not found. You must specifiy the URL to a Blazemeter test to run." > /dev/stderr
```
Comment (tisba) [inline blazemeter-cloud/docker-entrypoint.sh:15]:
```suggestion
	echo "ERROR: BLAZEMETER_TEST_URL not found. You must specify the URL to a Blazemeter test to run." > /dev/stderr
```

File: blazemeter-cloud/README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# BlazeMeter Cloud Trial Image
+
+The [BlazeMeter Cloud](https://www.blazemeter.com/) trial job enables you to use BlazeMeter performance tests in your Optimize Pro experiments. It leverages a custom container image, based on the the official BlazeMeter Taurus([https://](https://gettaurus.org/) [image](https://hub.docker.com/r/blazemeter/taurus/).
```
Comment (tisba) [inline blazemeter-cloud/README.md:3]: The link looks off. Maybe like this?

```suggestion
The [BlazeMeter Cloud](https://www.blazemeter.com/) trial job enables you to use BlazeMeter performance tests in your Optimize Pro experiments. It leverages a custom container image, based on the the official BlazeMeter ([Taurus](https://gettaurus.org/) [image](https://hub.docker.com/r/blazemeter/taurus/).
```

File: blazemeter-cloud/docker-entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+set -e
+
+if [ -z "${BLAZEMETER_API_ID}" ]; then
+	echo "ERROR: BLAZEMETER_API_ID not found. An API ID and an API Secret are required to access Blazemeter." > /dev/stderr
```
Comment (tisba) [inline blazemeter-cloud/docker-entrypoint.sh:5]:
```suggestion
	echo "ERROR: BLAZEMETER_API_ID not found. An API ID and an API Secret are required to access BlazeMeter cloud." > /dev/stderr
```

File: blazemeter-cloud/docker-entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	exit 1
+fi
+
+if [ -z "${BLAZEMETER_API_SECRET}" ]; then
+	echo "ERROR: BLAZEMETER_API_SECRET not found. An API ID and an API Secret are required to access Blazemeter." > /dev/stderr
```
Comment (tisba) [inline blazemeter-cloud/docker-entrypoint.sh:10]:
```suggestion
	echo "ERROR: BLAZEMETER_API_SECRET not found. An API ID and an API Secret are required to access BlazeMeter cloud." > /dev/stderr
```

File: blazemeter-cloud/docker-entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+	exit 1
+fi
+
+if [ -z "${BLAZEMETER_TEST_URL}" ]; then
+	echo "ERROR: BLAZEMETER_TEST_URL not found. You must specify the URL to a Blazemeter test to run." > /dev/stderr
```
Comment (tisba) [inline blazemeter-cloud/docker-entrypoint.sh:15]:
```suggestion
	echo "ERROR: BLAZEMETER_TEST_URL not found. You must specify the URL to a BlazeMeter cloud test to run." > /dev/stderr
```

**Review by tisba — COMMENTED** [review]
Just some nit-picky details 🙈 

Oh and one incorrect link title 😁

File: blazemeter-cloud/README.md (hunk 2 above)
Comment (ztreinhart) [reply blazemeter-cloud/README.md]: Fixed in https://github.com/thestormforge/optimize-trials/pull/17/commits/9d1da3074b8c5d105ddd8a948898ae079c522f47

**Pull request merged**